app/command_giphy.go

```diff
@@ -0,0 +1,255 @@
+			} `json:"preview_webp"`
+			FixedWidthStill struct {
+				URL    string `json:"url"`
+				Width  string `json:"width"`
+				Height string `json:"height"`
+			} `json:"fixed_width_still"`
+			FixedWidthSmall struct {
+				URL      string `json:"url"`
+				Width    string `json:"width"`
+				Height   string `json:"height"`
+				Size     string `json:"size"`
+				Mp4      string `json:"mp4"`
+				Mp4Size  string `json:"mp4_size"`
+				Webp     string `json:"webp"`
+				WebpSize string `json:"webp_size"`
+			} `json:"fixed_width_small"`
+			DownsizedSmall struct {
+				Width   string `json:"width"`
+				Height  string `json:"height"`
+				Mp4     string `json:"mp4"`
+				Mp4Size string `json:"mp4_size"`
+			} `json:"downsized_small"`
+			FixedWidthDownsampled struct {
+				URL      string `json:"url"`
+				Width    string `json:"width"`
+				Height   string `json:"height"`
+				Size     string `json:"size"`
+				Webp     string `json:"webp"`
+				WebpSize string `json:"webp_size"`
+			} `json:"fixed_width_downsampled"`
+			DownsizedMedium struct {
+				URL    string `json:"url"`
+				Width  string `json:"width"`
+				Height string `json:"height"`
+				Size   string `json:"size"`
+			} `json:"downsized_medium"`
+			Original struct {
+				URL      string `json:"url"`
+				Width    string `json:"width"`
+				Height   string `json:"height"`
+				Size     string `json:"size"`
+				Frames   string `json:"frames"`
+				Mp4      string `json:"mp4"`
+				Mp4Size  string `json:"mp4_size"`
+				Webp     string `json:"webp"`
+				WebpSize string `json:"webp_size"`
+			} `json:"original"`
+			FixedHeight struct {
+				URL      string `json:"url"`
+				Width    string `json:"width"`
+				Height   string `json:"height"`
+				Size     string `json:"size"`
+				Mp4      string `json:"mp4"`
+				Mp4Size  string `json:"mp4_size"`
+				Webp     string `json:"webp"`
+				WebpSize string `json:"webp_size"`
+			} `json:"fixed_height"`
+			Looping struct {
+				Mp4     string `json:"mp4"`
+				Mp4Size string `json:"mp4_size"`
+			} `json:"looping"`
+			OriginalMp4 struct {
+				Width   string `json:"width"`
+				Height  string `json:"height"`
+				Mp4     string `json:"mp4"`
+				Mp4Size string `json:"mp4_size"`
+			} `json:"original_mp4"`
+			PreviewGif struct {
+				URL    string `json:"url"`
+				Width  string `json:"width"`
+				Height string `json:"height"`
+				Size   string `json:"size"`
+			} `json:"preview_gif"`
+		} `json:"images"`
+	} `json:"data"`
+	Meta struct {
+		Status     int    `json:"status"`
+		Msg        string `json:"msg"`
+		ResponseID string `json:"response_id"`
+	} `json:"meta"`
+}
+
+func (me *GiphyProvider) DoCommand(args *model.CommandArgs, message string) *model.CommandResponse {
+	// TODO: this is a developer API key that cannot be used in production
+	// we have to upgrade it to a production key before releasing this feature
+	// see https://developers.giphy.com/docs/
+	giphyApiKey := "*****"
+
+	escapedQuery := url.PathEscape(message)
```

Comment:
`url.QueryEscape` would be the function to use here. `url.PathEscape` doesn't escape several characters such as `&` and `=`.